samples/net/coap_server/README.rst

```diff
@@ -1,4 +1,4 @@
-.. _zoap-server-sample:
+.. _coap-server-sample:
```

Comment:
There's a reference to zoap-server-sample in doc/subsystems/networking/overview.rst:66 that is broken with this change (does overview.rst need to be changed to reference coap-server-sample now?)